docs/Contributing.adoc

```diff
@@ -9,13 +9,13 @@ These are our contribution guidelines for helping out with the project. Any sugg
 * The Ansible link:https://docs.ansible.com/ansible/latest/community/code_of_conduct.html[Code of Conduct] still applies.
 * For git messages, branch names, etc., follow link:https://github.com/redhat-cop/agnosticd/blob/development/docs/git-style-guide.adoc[Git Style Guide].
 * Pull Requests should reference an *issue* whenever possible.
-** Completely new functionality e.g. a new role or new config etc, must reference an issue. 
+** Completely new functionality e.g. a new role or new config etc, must reference an issue.
```

Comment:
Hm. Do we really want to mandate that?

Reply:
I agree...After thinking about this, I am not for it. I think it should be the opposite. If you are touching, fixing, changing, an *existing* config or role, you should need an issue. Completely new roles or configs should not need an issue, but a well crafted description. Almost no one does this and I'm starting to ask more people to get in the habit.

Now, with changes to existing configs/roles that require an issue...that can get tedious, but we can't break the rules just because someone says "oh this is just one thing no big deal". Issues for changes are either enforced or they are not. Is having a well crafted description in the PR sufficient to negate the need for an issue? Do we make people copy and paste that description to an issue just to reference it? That is busy work. No thanks.

I think there will be some nuance here.

Add - completely new role or config. No issue needed. PR must have well crafted description with enough detail for someone to understand exactly what the role or config is doing. It must also follow other rules.

Change - changing an existing role as an elective or non-essential improvement. Does this need an issue?

Fix - How is this different than change? Is it break/fix? If anything needs an issue open, it would be Fix, but this seems very subjective to whoever is submitting.

Remove - Again, a well crafted description should explain why a config/role is being removed. Of course, I have yet to see this unicorn in the wild, so probably the lease important of the three :)

I reckon this comes down to - what is forcing the opening of issues really doing for any of these? I'm not against using them when people want to report an issue, but considering the speed at which things are reported and must be fixed I think it is going to become an inconvenient step *at this time* to force it.

Reply:
I agree we usually don't provide an issue and a PR with a good explanation and description can be self-sufficient.
i pushed a change. Let me know what you think now.